### PR TITLE
Feature/oh my class/oh my backend#42 improve docker file

### LINKED ISSUE #42: Improve docker file
Auto create database on docker-compose up, if there is none.

### PR DESCRIPTION
**Pull request by Z-100**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Only needed on windows: Replace with absolute path to your project
+ABSOLUTE_PATH_TO_INIT=[PATH-TO]oh-my-backend
```
Comment (Lezurex) [inline .env:2]: Why or rather where is this needed?

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,6 @@ target/
 build/
 !**/src/main/**/build/
 !**/src/test/**/build/
+
+### Custom ###
+*.env
```
Comment (Lezurex) [inline .gitignore]: Custom .env files are usually called .env.local, .env.test, etc. So the pattern should be .env.*

File: data/application/ohmyclass-init.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+CREATE DATABASE IF NOT EXISTS `ohmyclass` DEFAULT CHARACTER SET utf8 COLLATE utf8_general_ci;
```
Comment (Lezurex) [inline data/application/ohmyclass-init.sql]: Do we even need this, if we have the **MYSQL_DATABASE** var on line 14 in the docker-compose.yml?

File: docker-compose.yml [hunk 4]
```diff
@@ -5,20 +5,28 @@ version: '3.1'
...
+      MYSQL_DATABASE: ohmyclass
     volumes:
       - db:/var/lib/mysql
+    command: --init-file ./data/application/ohmyclass-init.sql
 
   phpmyadmin:
-    image: phpmyadmin
+    image: phpmyadmin/phpmyadmin:latest
+    container_name: phpmyadmin
+    restart: always
```
Comment (Lezurex) [inline docker-compose.yml:21]: This thing is annoying. It'll always restart when the Docker daemon starts.

File: docker-compose.yml [hunk 5]
```diff
@@ -5,20 +5,28 @@ version: '3.1'
...
+    image: mariadb:10.9.3
     ports:
       - 3307:3306
     environment:
+      MYSQL_ROOT_USER: root
       MYSQL_ROOT_PASSWORD: root
+      MYSQL_DATABASE: ohmyclass
     volumes:
       - db:/var/lib/mysql
+    command: --init-file ./data/application/ohmyclass-init.sql
```
Comment (Lezurex) [inline docker-compose.yml]: See ohmyclass-init.sql.

File: docker-compose.yml [hunk 6]
```diff
@@ -5,20 +5,28 @@ version: '3.1'
...
 
   phpmyadmin:
-    image: phpmyadmin
+    image: phpmyadmin/phpmyadmin:latest
```
Comment (Lezurex) [inline docker-compose.yml]: Why change to an older version?

**Review by Lezurex — CHANGES_REQUESTED** [review]
I'll submit a commit shortly.

File: .env (hunk 1 above)
Comment (Z-100) [reply .env:2]: comment on line 1

**Comment by Z-100** [comment]
@Lezurex Fix it. I have no idea of docker

File: .env (hunk 1 above)
Comment (Lezurex) [reply .env:2]: I read that, but I don't see where this environment variable is being used ...

**Review by Lezurex — APPROVED** [review]

**Pull request merged**